src/esp/gfx/RenderingTarget.cpp

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#include <Magnum/GL/Buffer.h>
+#include <Magnum/GL/DefaultFramebuffer.h>
+#include <Magnum/GL/Framebuffer.h>
+#include <Magnum/GL/Renderbuffer.h>
+#include <Magnum/GL/RenderbufferFormat.h>
+#include <Magnum/GL/Renderer.h>
+#include <Magnum/GL/Texture.h>
+#include <Magnum/GL/TextureFormat.h>
+#include <Magnum/Image.h>
+#include <Magnum/PixelFormat.h>
+
+#include "RenderingTarget.h"
+#include "magnum.h"
+
+using namespace Magnum;
+
+namespace esp {
+namespace gfx {
+struct RenderingTarget::Impl {
+  Impl(WindowlessContext::ptr context, int height, int width)
+      : context_{context},
+        framebufferSize_(width, height),
+        colorBuffer_(),
+        depthBuffer_(),
+        objectIdBuffer_(),
+        depthRenderbuffer_(),
+        framebuffer_(Magnum::NoCreate) {
+    colorBuffer_.setStorage(GL::RenderbufferFormat::SRGB8Alpha8,
+                            framebufferSize_);
+    depthBuffer_.setStorage(GL::RenderbufferFormat::R32F, framebufferSize_);
+    objectIdBuffer_.setStorage(GL::RenderbufferFormat::R32UI, framebufferSize_);
+    depthRenderbuffer_.setStorage(GL::RenderbufferFormat::Depth24Stencil8,
+                                  framebufferSize_);
+
+    framebuffer_ = GL::Framebuffer{{{}, framebufferSize_}};
+    framebuffer_
+        .attachRenderbuffer(GL::Framebuffer::ColorAttachment{0}, colorBuffer_)
+        .attachRenderbuffer(GL::Framebuffer::ColorAttachment{1}, depthBuffer_)
+        .attachRenderbuffer(GL::Framebuffer::ColorAttachment{2},
+                            objectIdBuffer_)
+        .attachRenderbuffer(GL::Framebuffer::BufferAttachment::Depth,
+                            depthRenderbuffer_)
+        .mapForDraw({{0, GL::Framebuffer::ColorAttachment{0}},
+                     {1, GL::Framebuffer::ColorAttachment{1}},
+                     {2, GL::Framebuffer::ColorAttachment{2}}});
+    CORRADE_INTERNAL_ASSERT(
+        framebuffer_.checkStatus(GL::FramebufferTarget::Draw) ==
+        GL::Framebuffer::Status::Complete);
+  }
+
+  void renderEnter() {
+    framebuffer_.bind();
+    framebuffer_.clear(GL::FramebufferClear::Color |
+                       GL::FramebufferClear::Depth);
+    framebuffer_.clearColor(1, Vector4{});
+    framebuffer_.clearColor(2, Vector4ui{});
+    framebuffer_.bind();
+  }
+
+  void renderExit() { glBindFramebuffer(GL_FRAMEBUFFER, 0); }
```

Comment:
`GL::defaultFramebuffer.bind()` instead. Calling `glBind*()` directly messes up with Magnum's state tracker and you'd need to call `GL::Context::resetState()` to avoid unpredictable behavior.

Reply:
Whoops!  Looks like I forgot to put a fixme to find the correct Magnum API :)

Reply:
@mosra -- Is there any reason why I should set the frame-buffer back to default?  It actually has a non-trivial impact on performance to do so for some reason.

Reply:
I don't know, you tell me, it was your code at first :) If I see correctly, the previous implementation didn't have this?

Switching framebuffers has always a big perf impact, yes, so ideally you want to have as little of those as possible. Unless something else later in the code *expects* that the default framebuffer is bound, I would say switching back is not needed. Or there could be an explicit binding of the default framebuffer later on the python side.

Thinking further -- because the framebuffers differ *only* in their size, why not have just one global framebuffer (like you had previously), that's as large as the largest sensor and then render / read just a part of it? All Magnum APIs are able to work with framebuffer subranges, but not sure if GLTensor can.

Reply:
Yep, doesn't seem like there is a need to bind the default framebuffer :)

RE 1 big framebuffer:  This is something that I had considered, but the ownership problem for CUDA-GL interop becomes difficult to manage (i.e. who register a given renderbuffer with CUDA-GL).  CUDA-GL also doesn't directly support reading only part of a framebuffer so we'd have to do the slicing ourselves.